### DISCUSSION TITLE
Support color opacity with CSS variable

### DISCUSSION THREAD

**Discussion by wdavidw** [tailwindlabs/tailwindcss#9883] [Ideas]
It seems like CSS opacity is not supported when the color is obtained from a CSS variable. Here is an example you can copy/paste into [Tailwind Play](https://play.tailwindcss.com/):

```html
<div class="absolute w-full h-full bg-slate-300 [--test:#00FF00]">
  <div class="h-[33%] bg-[#00FF00]/50">
    work
  </div>
  <div class="h-[33%] bg-[var(--test)]/50">
    does not work
  </div>
  <div class="h-[33%] bg-[var(--test)]">
    work
  </div>
</div>
```

**Comment by adamwathan** [comment]
Hey! You can do this but you have to set things up a little differently, see the documentation on this topic here:

https://tailwindcss.com/docs/customizing-colors#using-css-variables

**Reply by wdavidw** [reply]
Thank you @adamwathan, I am aware of this approach. It adds extra complexity and it is not what we would expect.

However, having a second look at `bg-[var(--test)]/50`, it seems like it is not possible for Tailwind to compile into CSS. It is technically not implementable, right ?

**Reply by adamwathan** [reply]
Yeah the way outlined in the docs is the only way to make it even possible with CSS. Currently CSS doesn’t offer any way to adjust the opacity channel of a color once that color is sort of “locked in” to a variable, they’d need to add some sort of color function to make it possible like:

```css
:root {
  --my-color: rgb(76, 98, 177);
}

.my-class {
  background-color: alpha(var(--my-color), 0.4);
}
```

…but nothing like that currently exists and isn’t possible to polyfill at build time since it needs to be able to happen at runtime in the browser.

**Comment by k16e** [comment]
I stumbled on this from search today, 2 years later. Would be really nice if we got a w/around.

**Comment by phaux** [comment]
Works in v4

```html
<div class="[--accent:hotpink]">
  <p class="text-(--accent)/50">Test</p>
</div>
```

https://play.tailwindcss.com/bf1YQyljxU

**Reply by k16e** [reply]
v4 is probably tw's best yet... then again, shouldn't it only get better? Thanks

**Reply by ivan-kleshnin** [reply]
@k16e-me

Is there a convenient way to reproduce this:

```css
/* CSS file */
theme(--color-slate-500 / 10%)
```

with `var`? 

- `opacity`, separate rule -- isn't enough for gradients and other multi-color settings
- `color-mix(in srgb, var(--my-hex-color) calc(<alpha-value> * 100%), transparent)` -- too long, cumbersome
- `rgb(var(--color-primary), <alpha-value>)` -- rounds down to more limited RGB space

**Reply by wongjn** [reply]
You can use [`--alpha()`](https://tailwindcss.com/docs/functions-and-directives#alpha-function): https://play.tailwindcss.com/8oRCGVsCUV?file=css
